slave/brutedict/brutedict.go

```diff
@@ -0,0 +1,53 @@
+package brutedict
+
+import (
+	"math/big"
+
+	"github.com/aaronang/cong-the-ripper/lib"
+)
+
+type BruteDict struct {
+	task  *lib.Task
+	queue chan []byte
+	quit  chan bool
+}
+
+func New(task *lib.Task) (bd *BruteDict) {
+	bd = &BruteDict{
+		task:  task,
+		queue: make(chan []byte),
+		quit:  make(chan bool),
+	}
+
+	go bd.list(task.Alphabet, task.Start, task.TaskLen)
+	return
+}
+
+func (bd *BruteDict) Next() (candidate []byte) {
+	select {
+	case candidate = <-bd.queue:
+	case <-bd.quit:
+	}
+	return
+}
+
+func (bd *BruteDict) Close() {
+	close(bd.queue)
+}
+
+func (bd *BruteDict) list(alph lib.Alphabet, start []byte, length int64) {
+	len := len(start)
+	currentComb := lib.BytesToBigInt(alph, start)
+
+	for length > 0 {
+		byteArray, overflow := lib.BigIntToBytes(alph, currentComb, len)
+		if overflow {
+			break
+		}
+		bd.queue <- lib.ReverseArray(byteArray)
```

Comment:
I think we need some benchmark for this operation and see how it affects the hashrate. Not sure if my thought process makes any sense, but this might have a significant impact on the hashrate. Correct me if I am wrong. I guess we can make an issue for this if you guys agree. If not, just ignore this comment :smile: